app/javascript/components/notifications/icon.jsx

```diff
@@ -0,0 +1,38 @@
+import React, { useReducer, useEffect } from 'react'
+import consumer from '../../application/action_cable_consumer'
+
+function reducer(state, action) {
+  switch (action.type) {
+    case 'notifications.changed':
+      return { count: action.payload.count }
+  }
+}
+
+export function NotificationIcon({ count }) {
+  const [state, dispatch] = useReducer(reducer, { count: count })
+
+  function isUnread() {
+    return state.count > 0
+  }
+
+  useEffect(() => {
```

Comment:
Absolutely fine for now, probably want to extract this into a custom hook at some point, when you start getting more of these subscriptions.